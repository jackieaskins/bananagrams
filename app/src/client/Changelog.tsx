import {
  Container,
  Heading,
  Table,
  TableContainer,
  Tbody,
  Td,
  Th,
  Thead,
  Tr,
} from "@chakra-ui/react";

type Change = {
  id: string;
  date: string;
  description: string;
};

const DATE_FORMATTER = new Intl.DateTimeFormat("en-us", {
  dateStyle: "medium",
  timeZone: "UTC",
});

const MAJOR_CHANGES: Change[] = [
  {
    id: "redesign",
    date: "2024-01-18",
    description:
      "Gameplay has been completely redesigned! Give it a try, report bugs, and provide feedback.",
  },
  {
    id: "swap-tiles",
    date: "2023-12-02",
    description:
      "Added the ability to swap tiles on the board by placing one tile on top of another.",
  },
  {
    id: "leave-game-tiles",
    date: "2023-11-30",
    description:
      "Tiles will no longer return to the bunch when a player leaves the game.",
  },
  {
    id: "game-start",
    date: "2023-11-25",
    description:
      "Added start game button. The game will no longer automatically start when all players are ready.",
  },
  {
    id: "spectator-view",
    date: "2023-11-24",
    description:
      "Added spectator mode. Players can now join in-progress games as spectators!",
  },
  {
    id: "changelog",
    date: "2023-11-17",
    description: "Added changelog.",
  },
  {
    id: "dark-mode",
    date: "2023-11-11",
    description: "Added dark mode.",
  },
  {
    id: "and-we-back",
    date: "2023-11-10",
    description: "We're back!",
  },
].map(({ date, ...rest }) => ({
  date: DATE_FORMATTER.format(new Date(date)),
  ...rest,
}));

export default function Changelog(): JSX.Element {
  return (
    <Container>
      <Heading as="h1" textAlign="center" marginY="35px">
        Changelog
      </Heading>

      <TableContainer>
        <Table variant="striped">
          <Thead>
            <Tr>
              <Th textAlign="right">Date</Th>
              <Th>Description</Th>
            </Tr>
          </Thead>

          <Tbody>
            {MAJOR_CHANGES.map(({ id, date, description }) => (
              <Tr key={id}>
                <Td
                  textAlign="right"
                  overflowWrap="break-word"
                  verticalAlign="top"
                >
                  {date}
                </Td>
                <Td whiteSpace="normal">{description}</Td>
              </Tr>
            ))}
          </Tbody>
        </Table>
      </TableContainer>
    </Container>
  );
}
