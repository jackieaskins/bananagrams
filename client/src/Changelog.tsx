import {
  Container,
  Heading,
  Table,
  TableContainer,
  Tbody,
  Td,
  Th,
  Thead,
  Tr,
} from "@chakra-ui/react";

type Change = {
  id: string;
  date: string;
  description: string;
};

const DATE_FORMATTER = new Intl.DateTimeFormat("en-us", {
  dateStyle: "medium",
});

const MAJOR_CHANGES: Change[] = [
  {
    id: "changelog",
    date: "2023-11-17",
    description: "Added changelog",
  },
  {
    id: "dark-mode",
    date: "2023-11-11",
    description: "Added dark mode",
  },
  {
    id: "and-we-back",
    date: "2023-11-10",
    description: "We're back!",
  },
].map(({ date, ...rest }) => ({
  date: DATE_FORMATTER.format(new Date(date)),
  ...rest,
}));

export default function Changelog(): JSX.Element {
  return (
    <Container>
      <Heading as="h1" textAlign="center" marginY="35px">
        Changelog
      </Heading>

      <TableContainer>
        <Table variant="striped">
          <Thead>
            <Tr>
              <Th textAlign="right">Date</Th>
              <Th>Description</Th>
            </Tr>
          </Thead>

          <Tbody>
            {MAJOR_CHANGES.map(({ id, date, description }) => (
              <Tr key={id}>
                <Td textAlign="right">{date}</Td>
                <Td>{description}</Td>
              </Tr>
            ))}
          </Tbody>
        </Table>
      </TableContainer>
    </Container>
  );
}
