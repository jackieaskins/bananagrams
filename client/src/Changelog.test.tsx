import { shallow } from "enzyme";
import Changelog from "./Changelog";

describe("<Changelog />", () => {
  it("renders table of changes", () => {
    // NOTE: Snapshot will have dates one day behind...
    expect(shallow(<Changelog />)).toMatchSnapshot();
  });
});
