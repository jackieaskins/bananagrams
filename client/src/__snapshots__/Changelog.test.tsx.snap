// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Changelog /> renders table of changes 1`] = `
<Container>
  <Heading
    as="h1"
    marginY="35px"
    textAlign="center"
  >
    Changelog
  </Heading>
  <ForwardRef>
    <Table
      variant="striped"
    >
      <ForwardRef>
        <ForwardRef>
          <ForwardRef
            textAlign="right"
          >
            Date
          </ForwardRef>
          <ForwardRef>
            Description
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
      <ForwardRef>
        <ForwardRef
          key="spectator-view"
        >
          <ForwardRef
            textAlign="right"
          >
            Nov 23, 2023
          </ForwardRef>
          <ForwardRef>
            Added spectator mode. Players can now join in-progress games as a spectator!
          </ForwardRef>
        </ForwardRef>
        <ForwardRef
          key="changelog"
        >
          <ForwardRef
            textAlign="right"
          >
            Nov 16, 2023
          </ForwardRef>
          <ForwardRef>
            Added changelog
          </ForwardRef>
        </ForwardRef>
        <ForwardRef
          key="dark-mode"
        >
          <ForwardRef
            textAlign="right"
          >
            Nov 10, 2023
          </ForwardRef>
          <ForwardRef>
            Added dark mode
          </ForwardRef>
        </ForwardRef>
        <ForwardRef
          key="and-we-back"
        >
          <ForwardRef
            textAlign="right"
          >
            Nov 9, 2023
          </ForwardRef>
          <ForwardRef>
            We're back!
          </ForwardRef>
        </ForwardRef>
      </ForwardRef>
    </Table>
  </ForwardRef>
</Container>
`;
